src/AltinnCore/Common/Services/Implementation/DataStudioSI.cs

```diff
@@ -230,51 +253,53 @@ public Task<List<AttachmentList>> GetFormAttachments(string org, string app, int
         }
 
         /// <inheritdoc />
-        public void DeleteFormAttachment(string org, string app, int instanceOwnerId, Guid instanceId, string attachmentType, string attachmentId)
+        public Task<bool> DeleteFormAttachment(string org, string app, int instanceOwnerId, Guid instanceGuid, Guid dataGuid)
         {
             string developer = AuthenticationHelper.GetDeveloperUserName(_httpContextAccessor.HttpContext);
             string testDataForParty = _settings.GetTestdataForPartyPath(org, app, developer);
-            string instanceFilePath = $"{testDataForParty}{instanceOwnerId}/{instanceId}/{instanceId}.json";
+            string instanceFilePath = $"{testDataForParty}{instanceOwnerId}/{instanceGuid}/{instanceGuid}.json";
 
-            lock (Guard(instanceId))
+            lock (Guard(instanceGuid))
             {
                 string instanceData = File.ReadAllText(instanceFilePath);
 
                 Instance instance = JsonConvert.DeserializeObject<Instance>(instanceData);
-                DataElement removeFile = instance.Data.Find(m => m.Id == attachmentId);
+                DataElement removeFile = instance.Data.Find(m => m.Id == dataGuid.ToString());
 
                 instance.Data.Remove(removeFile);
 
                 string instanceDataAsString = JsonConvert.SerializeObject(instance);
                 File.WriteAllText(instanceFilePath, instanceDataAsString);
             }
 
-            string pathToDelete = $"{_settings.GetTestdataForPartyPath(org, app, developer)}{instanceOwnerId}/{instanceId}/data/{attachmentId.AsFileName()}";
+            string pathToDelete = $"{_settings.GetTestdataForPartyPath(org, app, developer)}{instanceOwnerId}/{instanceGuid}/data/{dataGuid.ToString().AsFileName()}";
             File.Delete(pathToDelete);
+
+            return Task.FromResult(true);
         }
 
         /// <inheritdoc />
-        public async Task<Guid> SaveFormAttachment(string org, string app, int instanceOwnerId, Guid instanceId, string attachmentType, string attachmentName, HttpRequest request)
+        public async Task<DataElement> SaveFormAttachment(string org, string app, int instanceOwnerId, Guid instanceGuid, string attachmentType, string attachmentName, HttpRequest request)
```

Comment:
my suggestion would be to remove guid from the name either as part of this PR or as a new issue

Reply:
what would be the alternative name? I think instanceId would give the wrong associations..

Reply:
resolving as we don't have any alternative names as this point.